crates/miden-objects/src/account/component/code.rs

```diff
@@ -0,0 +1,47 @@
+use miden_assembly::Library;
+use miden_processor::MastForest;
+
+// ACCOUNT COMPONENT CODE
+// ================================================================================================
+
+/// A [`Library`] that has been assembled for use as component code.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct AccountComponentCode(Library);
```

Comment:
This is fine for now, but in the future, I think we could decompose the library into individual components and this could be something like:

```rust
pub struct AccountComponentCode {
    mast: Arc<MastForest>,
    exports: Vec<ProcedureExport>,
}
```

One of the reasons for this is that we are considering deprecating `Library` structs in favor of packages, and I'd like to reduce their use in general.